BookingsApi/BookingsApi/Controllers/JudiciaryPersonController.cs

```diff
@@ -0,0 +1,88 @@
+{
+    [Produces("application/json")]
+    [Route("judiciaryperson")]
+    [ApiController]
+    public class JudiciaryPersonController : Controller
+    {
+        private const string BulkItemErrorMessage = "Could not add or update external Judiciary user with External Id: {0}";
+        private readonly IQueryHandler _queryHandler;
+        private readonly ICommandHandler _commandHandler;
+        private readonly ILogger<JudiciaryPersonController> _logger;
+        
+        public JudiciaryPersonController(IQueryHandler queryHandler, ICommandHandler commandHandler, ILogger<JudiciaryPersonController> logger)
+        {
+            _queryHandler = queryHandler;
+            _commandHandler = commandHandler;
+            _logger = logger;
+        }
+        
+        [HttpPost]
+        [OpenApiOperation("BulkJudiciaryPersons")]
+        [ProducesResponseType((int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        [ProducesResponseType(typeof(BulkJudiciaryPersonResponse), (int)HttpStatusCode.OK)]
+        public async Task<IActionResult> BulkJudiciaryPersonsAsync(IEnumerable<JudiciaryPersonRequest> request)
+        {
+            var bulkResponse = new BulkJudiciaryPersonResponse();
+            
+            foreach (var item in request)
+            {
+                var validation = await new JudiciaryPersonRequestValidation().ValidateAsync(item);
+                if (!validation.IsValid)
+                {
+                    bulkResponse.ErroredRequests.Add(new JudiciaryPersonErrorResponse
+                    {
+                        Message = $"{string.Format(BulkItemErrorMessage, item.Id)} - {string.Join(", ", validation.Errors.Select(x => x.ErrorMessage))}", 
+                        JudiciaryPersonRequest = item
+                    });
+                    
+                    continue;
+                }
+            
+                try
+                {
+                    var query = new GetJudiciaryPersonByExternalRefIdQuery(item.Id);
+                    var judiciaryPerson = await _queryHandler.Handle<GetJudiciaryPersonByExternalRefIdQuery, JudiciaryPerson>(query);
+
+                    if (judiciaryPerson == null)
+                    {
+                        await _commandHandler.Handle(new AddJudiciaryPersonCommand(item.Id, item.PersonalCode, item.Title, item.KnownAs, item.Surname,
+                            item.Fullname, item.PostNominals, item.Email));
+                    }
+                    else
+                    {
+                        await _commandHandler.Handle(new UpdateJudiciaryPersonByExternalRefIdCommand(item.Id, item.PersonalCode, 
+                            item.Title, item.KnownAs, item.Surname, item.Fullname, item.PostNominals, item.Email));
+                    }
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, string.Format(BulkItemErrorMessage, item.Id));
```

Comment:
`.LogError(ex, BulkItemErrorMessage, item.Id);` will add the item id to custom dimensions and automatically format

Reply:
Good catch... DONE